### PR TITLE
Publish generated views and queries to a generated-sql branch

### LINKED ISSUE #1742: Implement `generated-sql` branch to capture missing datasets
Right now not every dataset is checked in to `sql/`, which results to missing content in the generated docs that I'm working on in #1735. 

[@jklukas's comment:](https://github.com/mozilla/bigquery-etl/pull/1735#issuecomment-771180105)

> For Glean applications, we have some machinery that generates and runs queries that don't appear under the sql folder that's checked in, so they don't appear in these generated docs. For example, there's a user-facing org_mozilla_ios_firefox dataset that's missing here.
> 
> This could be more motivation for maintaining a generated-sql branch (see #1110 (comment)), but shouldn't block this effort. This could be another future improvement.
>

### PR DESCRIPTION
**Pull request by jklukas**
Fixes #1742

The [`generated-sql`](https://github.com/mozilla/bigquery-etl/tree/generated-sql) branch now exists and you can browse the contents. See, for example, [telemetry.main](https://github.com/mozilla/bigquery-etl/tree/generated-sql/sql/moz-fx-data-shared-prod/telemetry/main).

Follow-ups for which I'll file issues:

- This doesn't currently publish the generated Glean baseline ETL queries
  and views; we'll need to update that logic to use probe-scraper metadata
  rather than listing tables in BigQuery (due to creds) to integrate it.
- Docs publishing should reference this generated content rather than the content on `master`

### CODE REVIEW

File: sql/moz-fx-data-shared-prod/org_mozilla_fenix_nightly/baseline_clients_daily/view.sql [hunk 1]
```diff
@@ -1,8 +0,0 @@
--- Generated via bigquery_etl.glean_usage
```
Comment (jklukas) [inline sql/moz-fx-data-shared-prod/org_mozilla_fenix_nightly/baseline_clients_daily/view.sql:1]: The existing checked-in content for `org_mozilla_fenix_nightly` is stale. In practice, it's ignored by the baseline ETL code in this repository. We remove it now to reduce confusion.

File: bigquery_etl/view/generate_stable_views.py [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+  * REPLACE(
+    {replacements})
+FROM
+  `{target}`
+"""
+
+VIEW_METADATA_TEMPLATE = """\
+# Generated by bigquery_etl.view.generate_stable_views
+---
+friendly_name: Historical Pings for ``{document_namespace}/{document_type}`
```
Comment (scholtzan) [inline bigquery_etl/view/generate_stable_views.py:47]:
```suggestion
friendly_name: Historical Pings for `{document_namespace}/{document_type}`
```

File: bigquery_etl/view/generate_stable_views.py [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+    target_file = target_dir / "view.sql"
+
+    if target_file.exists():
+        return
+
+    # Exclude doctypes maintained in separate projects.
+    if schema.bq_dataset_family.startswith("pioneer"):
+        return
+    if schema.bq_dataset_family.startswith("rally"):
+        return
```
Comment (scholtzan) [inline bigquery_etl/view/generate_stable_views.py:124]: Should we maintain some kind of `SKIP` list for these (similar to what we have been doing in `dryrun`)?

File: bigquery_etl/view/generate_stable_views.py [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+
+    full_source_id = f"{target_project}.{schema.stable_table}"
+    full_view_id = f"{target_project}.{schema.user_facing_view}"
+    replacements = ["mozfun.norm.metadata(metadata)" " AS metadata"]
```
Comment (scholtzan) [inline bigquery_etl/view/generate_stable_views.py:128]:
```suggestion
    replacements = ["mozfun.norm.metadata(metadata) AS metadata"]
```
Is there a reason for splitting these replacement strings?

File: bigquery_etl/view/generate_stable_views.py [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+                    SchemaFile(
+                        schema_id=schema["$schema"],
+                        bq_dataset_family=pipeline_meta["bq_dataset_family"],
+                        bq_table=pipeline_meta["bq_table"],
+                        document_namespace=document_namespace,
+                        document_type=document_type,
+                        document_version=version,
+                    )
+                )
+                "".rstrip()
```
Comment (scholtzan) [inline bigquery_etl/view/generate_stable_views.py:227]: Does this do anything?

File: .circleci/config.yml [hunk 6]
```diff
@@ -175,6 +167,42 @@ jobs:
             cd generated_docs/
             mkdocs gh-deploy \
               -m "[ci skip] Deployed {sha} with MkDocs version: {version}"
+  generate-sql:
+    docker: *docker
+    steps:
+      - checkout
+      - *skip_forked_pr
```
Comment (wlach) [inline .circleci/config.yml:174]: Would this ever be run from a forked PR?

File: bigquery_etl/view/generate_stable_views.py [hunk 7]
```diff
@@ -0,0 +1,251 @@
+"""
```
Comment (wlach) [inline bigquery_etl/view/generate_stable_views.py:1]: What do you think about adding this to the bqetl cli? See #1648

**Review by scholtzan — APPROVED** [review]
The general approach looks good to me. Just a few nits.

File: bigquery_etl/view/generate_stable_views.py [hunk 8]
```diff
@@ -0,0 +1,251 @@
...
+            and schema.bq_table == "metrics_v1"
+        ):
+            # todo: use mozfun udfs
+            replacements += [
+                "mozdata.udf.normalize_fenix_metrics"
+                "(client_info.telemetry_sdk_build, metrics)"
+                " AS metrics"
+            ]
+        if schema.bq_dataset_family == "firefox_desktop":
+            # Bug 1672191
```
Comment (wlach) [inline bigquery_etl/view/generate_stable_views.py:148]: Can you fill this out a bit more?

**Review by wlach — CHANGES_REQUESTED** [review]
This makes a lot of sense to me, just one question.

File: .circleci/config.yml (hunk 6 above)
Comment (jklukas) [reply .circleci/config.yml:174]: Good point. Will remove.

File: bigquery_etl/view/generate_stable_views.py (hunk 5 above)
Comment (jklukas) [reply bigquery_etl/view/generate_stable_views.py:227]: Sure doesn't. Will remove.

File: bigquery_etl/view/generate_stable_views.py (hunk 7 above)
Comment (jklukas) [reply bigquery_etl/view/generate_stable_views.py:1]: I will file that as an additional follow-up issue to this PR. This script will be used by ops logic and for now I'm attempting to maintain as similar an interface as possible to the existing `generate_views` script.

**Comment by jklukas** [comment]
I've addressed all comments. I'd like to defer integrating with `bqetl` for the scope of this PR.

File: bigquery_etl/view/generate_stable_views.py [hunk 9]
```diff
@@ -0,0 +1,255 @@
...
+    # Exclude doctypes maintained in separate projects.
+    for prefix in SKIP_PREFIXES:
+        if schema.bq_dataset_family.startswith(prefix):
+            return
+
+    full_source_id = f"{target_project}.{schema.stable_table}"
+    full_view_id = f"{target_project}.{schema.user_facing_view}"
+    replacements = ["mozfun.norm.metadata(metadata) AS metadata"]
+    if schema.schema_id == "moz://mozilla.org/schemas/glean/ping/1":
+        replacements += ["mozfun.norm.glean_ping_info(ping_info)" " AS ping_info"]
```
Comment (scholtzan) [inline bigquery_etl/view/generate_stable_views.py:134]:
```suggestion
        replacements += ["mozfun.norm.glean_ping_info(ping_info) AS ping_info"]
```

File: bigquery_etl/view/generate_stable_views.py [hunk 10]
```diff
@@ -0,0 +1,255 @@
...
+                " AS metrics"
+            ]
+        if schema.bq_dataset_family == "firefox_desktop":
+            # FOG does not provide an app_name, so we inject the one that
+            # people already associate with desktop Firefox per bug 1672191.
+            replacements += [
+                "'Firefox' AS normalized_app_name",
+            ]
+    elif schema.schema_id.startswith("moz://mozilla.org/schemas/main/ping/"):
+        replacements += ["mozdata.udf.normalize_main_payload(payload)" " AS payload"]
```
Comment (scholtzan) [inline bigquery_etl/view/generate_stable_views.py:158]:
```suggestion
        replacements += ["mozdata.udf.normalize_main_payload(payload) AS payload"]
```

**Review by scholtzan — APPROVED** [review]

**Review by Iinh — APPROVED** [review]
👍 I'll update the docs to reference this branch once this PR is merged.

**Comment by jklukas** [comment]
> 👍 I'll update the docs to reference this branch once this PR is merged.

See https://github.com/mozilla/bigquery-etl/issues/1778 ; it's going to involve a bit more CircleCI config changing to get the flow from generated-sql to docs working.

**Review by wlach — APPROVED** [review]

File: bigquery_etl/view/generate_stable_views.py (hunk 9 above)
Comment (jklukas) [reply bigquery_etl/view/generate_stable_views.py:134]: Missed this one. Will fix before merging.

**Pull request merged**